superset/db_engine_specs/presto.py

```diff
@@ -392,28 +392,49 @@ def update_impersonation_config(
 
     @classmethod
     def get_table_names(
-        cls, database: Database, inspector: Inspector, schema: Optional[str]
+        cls,
+        database: Database,
+        inspector: Inspector,
+        schema: Optional[str],
     ) -> List[str]:
-        tables = super().get_table_names(database, inspector, schema)
-        if not is_feature_enabled("PRESTO_SPLIT_VIEWS_FROM_TABLES"):
```

Comment:
I'm not quite sure why this undocumented feature flag existed. Why wouldn't one want to split views from tables as is the case with the other engine specifications.

Reply:
When I added the feature at Lyft it would take 90 seconds to read all the views, so I thought it would be better to put it behind a feature flag.

Reply:
Yikes. @betodealmeida I'm somewhat surprised that is the case as per Trino's definition of their `get_table_names` method (which differs from PyHive) it uses very [similar logic](https://github.com/trinodb/trino-python-client/blob/master/trino/sqlalchemy/dialect.py#L194-L206) as what you provided for the `get_view_names`. i.e., querying a table within the `INFORMATION_SCHEMA` schema however your method should be faster given that `INFORMATION_SCHEMA.TABLES` is a superset of `INFORMATION_SCHEMA.VIEWS`.

I wonder if the poor performance was related to when the schema was undefined? This functionally, i.e., fetching all tables/views across all schemas, was deprecated in https://github.com/apache/superset/pull/21284 given that the SQL Lab frontend no longer supports searching across all schemas, i.e., a schema must be defined.